transport.go

```diff
@@ -2,20 +2,56 @@ package transport
 
 import (
 	"context"
+	"io"
 	"net"
 	"time"
 
 	logging "github.com/ipfs/go-log"
+	smux "github.com/libp2p/go-stream-muxer"
 	ma "github.com/multiformats/go-multiaddr"
-	manet "github.com/multiformats/go-multiaddr-net"
 )
 
 var log = logging.Logger("transport")
 
 // Conn is an extension of the net.Conn interface that provides multiaddr
 // information, and an accessor for the transport used to create the conn
 type Conn interface {
-	manet.Conn
```

Comment:
Are we still planning on using this interface or deprecating it entierly? If we still want to ensure we meet this interface, it would probably be a good idea to assert that `SingleStreamConn` conforms to it.